llvm/cmake/config-ix.cmake

```diff
@@ -157,7 +157,14 @@ if(LLVM_ENABLE_ZSTD)
     find_package(zstd QUIET)
   endif()
 endif()
+
+# If LLVM_USE_STATIC_ZSTD is specified, make sure we enable zstd only if static
+# libraries are found.
```

Comment:
I'll create a bug report and PR to upstream this change, if approved.

Reply:
For upstreaming:
issue: https://github.com/llvm/llvm-project/issues/113583
PR: https://github.com/llvm/llvm-project/pull/113584